### ISSUE TITLE
mmcblk* devices no serial found

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#757]
Some sdcard drivers produce block devices of /dev/mmcbl*
```
lsblk 
...
mmcblk0     179:0    0   7.5G  0 disk 
└─mmcblk0p1 179:1    0   7.5G  0 part 
```
Rockstor's primary block device serial number discovery mechanism uses:- /usr/bin/lsblk -P -o SERIAL
which as demonstrated fails to find the serial number of this device:-
```
/usr/bin/lsblk -P -o SERIAL /dev/mmcblk0
SERIAL=""
SERIAL=""
```
This property is shared with virtio based drives whose's serial could be found by either:-
```
udevadm info /sys/class/block/vda
or
cat /sys/block/vda/serial
```
as detailed in #673 and #675 respectively.

This mmcblk* device however doesn't have its serial accessible via the method currently implemented for virtio (though without being specific to virtio device types) ie there is no:-
/sys/block/mmcblk0/serial
but there is a:-
/sys/block/mmcblk0/device/serial
which cat returns the serial for successfully, however this path does not also exist for the virtio device.

So we can either extend the current implementation of get_virtio_disk_serial(device_name) in osi.py to check for both or revert to the originally proposed method for virtio devices ie that of 
"udevadm info" as it appears to works for both devices equally well and similarly ie:-

```
# virtio device:-
udevadm info /sys/class/block/vda | grep SERIAL
E: ID_SERIAL=fake-serial00001

# and the same command and path for
# mmcblk0 device
udevadm info /sys/class/block/mmcblk0 | grep SERIAL
E: ID_SERIAL=0x1ceff260
```

I propose a rewrite and re-name of get_virtio_disk_serial(device_name) to be more generic and employ udevadm which abstracts the serial number access mechanism and works equally for both virtio and these mmcblk* sdcard block devices.
There is greater complexity in parsing but the mechanism is shared and the call is infrequent.

@schakrava Do you agree?

**Comment by phillxnet** [comment]
Linking to related forum post where lsblk fails to retrieve unique serial numbers of drives attached to an Areca ARC1680IX ver 2.3 raid controller but udevadm does.

I am doing final testing on a pull request to resolve this issue and it should also address (via minor editing) the linked non unique serial number issue in the forum thread referenced.  The intention is to replace the existing lsblk serial retrieval with the proposed mechanism involving udevadm after further field testing using the minor editing of an incorporated flag.  Care has been taken to return the same serial as was returned by lsblk and as is commonly displayed on device labels.

**Comment by phillxnet** [comment]
Pull request #786  addresses this issue and is ready for review.  It also includes an off by default mechanism to over ride all serial number fetches as udevadm with the proposed simple cascade also returns serials the same as lsblk only more reliably and across more varied hardware.

**Comment by phillxnet** [comment]
Closing as fixed by #786 merged pull request.
